File: Domain/AppHistory.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class AppHistory
    {
		[Key]
      	[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
	  	public int Id { get; set; }
		public Guid AppDataId { get; set; }
		public string Action { get; set; }
		public int FromStage { get; set; }
		public int ToStage { get; set; }
		public int ActionBy { get; set; }
		public DateTime DateTime { get; set; }
		public string Comment { get; set; }
		public string Details1 { get; set; }
		public string Details2 { get; set; }
		public string Details3 { get; set; }
		public string Details4 { get; set; }
		public string Details5 { get; set; }
    }
}

File: client-app/src/features/AppHistory/AppHistory.ts
export interface IAppHistory {
	Id: number
	Action: string
	FromStage: number
	ToStage: number
	ActionBy: string
	DateTime: Date
	Comment: string
	Details1: string
	Details2: string
	Details3: string
	Details4: string
	Details5: string
}

export class AppHistory implements IAppHistory {
	Id: number = 0;
	Action: string = '';
	FromStage: number = 0;
	ToStage: number = 0;
	ActionBy: string = '';
	DateTime: Date = new Date() ;
	Comment: string = '';
	Details1: string = '';
	Details2: string = '';
	Details3: string = '';
	Details4: string = '';
	Details5: string = '';
  
  constructor(init?: IAppHistory) {
    Object.assign(this, init);
  }
}

